storage/innobase/include/sync0policy.h

```diff
@@ -105,7 +113,7 @@ class MutexDebug {
 
 			msg << m_mutex->policy().to_string();
 
-			if (os_thread_pf(m_thread_id) != ULINT_UNDEFINED) {
+			if (os_thread_pf(get_thread_id()) != ULINT_UNDEFINED) {
```

Comment:
Did it show up in TSAN report? If not either leave as is or protect whole function (first ```m_mutex```reference till ```return```) by new mutex.